### ISSUE TITLE
failed to parse compileSDKVersion ()

### ISSUE THREAD

**Issue by alshari86** [bitrise-steplib/steps-install-missing-android-tools#13]
It mentioned the issue fixed in https://github.com/bitrise-steplib/steps-install-missing-android-tools/issues/5
But I still got this problem when do Android build.

**Comment by viktorbenei** [comment]
Can you please copy paste the step's log here?

**Comment by alshari86** [comment]
(1) git-clone@3.4.3                                                          |
+------------------------------------------------------------------------------+
| id: git-clone                                                                |
| version: 3.4.3                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-05-12T18:16:27Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
[34mINFO[0m[18:16:27] Start installing (golang) with apt-get       
[34mINFO[0m[18:16:27]  * [32;1m[OK][0m Step dependency (go) installed, available. 

[34;1mGit Clone Configs:[0m
- CloneIntoDir: /bitrise/src
- RepositoryURL: git@bitbucket.org:ymedialabs/frb-android.git
[34;1mGit Checkout Configs:[0m
- Commit: 
- Tag: 
- Branch: test-ci
- CloneDepth: 
[34;1mGit Pull Request Configs:[0m
- PullRequestURI: 
- PullRequestID: 
- BranchDest: 
- PullRequestMergeBranch: 
- ResetRepository: No
[34;1mBitrise Build Configs:[0m
- BuildURL: https://www.bitrise.io/build/68384226801e764a
- BuildAPIToken: WUIKHfgu4poNgRrhgVOWVw

[34;1mGit clone repository[0m
=> git "init"
=> git "remote" "add" "origin" "git@bitbucket.org:ymedialabs/frb-android.git"
=> git "fetch"
=> git "checkout" "test-ci"
=> git "submodule" "update" "--init" "--recursive"
[34;1mExporting git logs[0m
=> GIT_CLONE_COMMIT_HASH
   value: ca49249fff224f8a69af03ba6797bb9654e5ab01

=> GIT_CLONE_COMMIT_MESSAGE_SUBJECT
   value: update target version

=> GIT_CLONE_COMMIT_MESSAGE_BODY
   value: 

=> GIT_CLONE_COMMIT_AUTHOR_NAME
   value: Umang Desai

=> GIT_CLONE_COMMIT_AUTHOR_EMAIL
   value: umang.desai@ymedialabs.com

=> GIT_CLONE_COMMIT_COMMITER_NAME
   value: Umang Desai

=> GIT_CLONE_COMMIT_COMMITER_EMAIL
   value: umang.desai@ymedialabs.com

[32;1mSuccess[0m
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1mgit-clone@3.4.3[0m                                               | 12 sec   |
+---+---------------------------------------------------------------+----------+

                                          ▼

+------------------------------------------------------------------------------+
| (2) Do anything with Script step                                             |
+------------------------------------------------------------------------------+
| id: script                                                                   |
| version: 1.1.3                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2017-05-12T18:16:39Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
Hello World!
+ echo 'Hello World!'
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1mDo anything with Script step[0m                                  | 1.19 sec |
+---+---------------------------------------------------------------+----------+

                                          ▼

+------------------------------------------------------------------------------+
| (3) install-missing-android-tools@1.0.2                                      |
+------------------------------------------------------------------------------+
| id: install-missing-android-tools                                            |
| version: 1.0.2                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-05-12T18:16:41Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
[34mINFO[0m[18:16:41] Start installing (golang) with apt-get       
[34mINFO[0m[18:16:41]  * [32;1m[OK][0m Step dependency (go) installed, available. 

[34;1mConfigs:[0m
- RootBuildGradleFile: banking/build.gradle
- GradlewPath: banking/gradlew
- UpdateSupportLibraryAndPlayServices: true
- AndroidHome: /opt/android-sdk-linux

[34;1mSearch for root settings.gradle file[0m
root build.gradle file: /bitrise/src/banking/build.gradle
root settings.gradle file:: /bitrise/src/banking/settings.gradle

[34;1mCollect build.gradle files to analyze[0m
active modules to analyze: [mobile misnapworkflow]
build.gradle files to analyze: [/bitrise/src/banking/mobile/build.gradle /bitrise/src/banking/misnapworkflow/build.gradle]

[34;1mCollect dependencies to ensure[0m
Analyze build.gradle file: /bitrise/src/banking/mobile/build.gradle
[33;1mFailed to parse build gradle file: /bitrise/src/banking/mobile/build.gradle, error: failed to determine compileSDKVersion, error: failed to find compileSdkVersion[0m
[33;1mSwitched to extended analyzer...[0m
[31;1mFailed to analyze build.gradle at: /bitrise/src/banking/mobile/build.gradle, error: failed to parse compileSDKVersion ()[0m
Analyze build.gradle file: /bitrise/src/banking/misnapworkflow/build.gradle
[33;1mFailed to parse build gradle file: /bitrise/src/banking/misnapworkflow/build.gradle, error: failed to determine compileSDKVersion, error: failed to find compileSdkVersion[0m
[33;1mSwitched to extended analyzer...[0m
[31;1mFailed to analyze build.gradle at: /bitrise/src/banking/misnapworkflow/build.gradle, error: failed to parse compileSDKVersion ()[0m

[34;1mEnsure dependencies[0m
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1minstall-missing-android-tools@1.0.2[0m                           | 3.48 sec |
+---+---------------------------------------------------------------+----------+

                                          ▼

+------------------------------------------------------------------------------+
| (4) change-workdir@1.0.1                                                     |
+------------------------------------------------------------------------------+
| id: change-workdir                                                           |
| version: 1.0.1                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2017-05-12T18:16:44Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
=> Changing working directory to /bitrise/src/banking
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1mchange-workdir@1.0.1[0m                                          | 1.21 sec |
+---+---------------------------------------------------------------+----------+

                                          ▼

+------------------------------------------------------------------------------+
| (5) gradle-runner@1.5.4                                                      |
+------------------------------------------------------------------------------+
| id: gradle-runner                                                            |
| version: 1.5.4                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-05-12T18:16:45Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
[34mINFO[0m[18:16:45] Start installing (golang) with apt-get       
[34mINFO[0m[18:16:45]  * [32;1m[OK][0m Step dependency (go) installed, available. 

[34;1mConfigs:[0m
  - GradlewPath: banking/gradlew
  - GradleFile: banking/build.gradle
  - GradleTasks: :mobile:assembleDebug
  - GradleOptions: --stacktrace
  - ApkFileIncludeFilter: *.apk
  - ApkFileExcludeFilter: *-unaligned.apk
  - MappingFileIncludeFilter: */mapping.txt
  - MappingFileExcludeFilter: 
  - DeployDir: /bitrise/deploy

[31;1mIssue with input: GradleFile not exist at: banking/build.gradle[0m

**Comment by viktorbenei** [comment]
Thanks for sharing @alshari86 !

The related step log:

```
+------------------------------------------------------------------------------+
| (3) install-missing-android-tools@1.0.2                                      |
+------------------------------------------------------------------------------+
| id: install-missing-android-tools                                            |
| version: 1.0.2                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2017-05-12T18:16:41Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
[34mINFO[0m[18:16:41] Start installing (golang) with apt-get       
[34mINFO[0m[18:16:41]  * [32;1m[OK][0m Step dependency (go) installed, available. 

[34;1mConfigs:[0m
- RootBuildGradleFile: banking/build.gradle
- GradlewPath: banking/gradlew
- UpdateSupportLibraryAndPlayServices: true
- AndroidHome: /opt/android-sdk-linux

[34;1mSearch for root settings.gradle file[0m
root build.gradle file: /bitrise/src/banking/build.gradle
root settings.gradle file:: /bitrise/src/banking/settings.gradle

[34;1mCollect build.gradle files to analyze[0m
active modules to analyze: [mobile misnapworkflow]
build.gradle files to analyze: [/bitrise/src/banking/mobile/build.gradle /bitrise/src/banking/misnapworkflow/build.gradle]

[34;1mCollect dependencies to ensure[0m
Analyze build.gradle file: /bitrise/src/banking/mobile/build.gradle
[33;1mFailed to parse build gradle file: /bitrise/src/banking/mobile/build.gradle, error: failed to determine compileSDKVersion, error: failed to find compileSdkVersion[0m
[33;1mSwitched to extended analyzer...[0m
[31;1mFailed to analyze build.gradle at: /bitrise/src/banking/mobile/build.gradle, error: failed to parse compileSDKVersion ()[0m
Analyze build.gradle file: /bitrise/src/banking/misnapworkflow/build.gradle
[33;1mFailed to parse build gradle file: /bitrise/src/banking/misnapworkflow/build.gradle, error: failed to determine compileSDKVersion, error: failed to find compileSdkVersion[0m
[33;1mSwitched to extended analyzer...[0m
[31;1mFailed to analyze build.gradle at: /bitrise/src/banking/misnapworkflow/build.gradle, error: failed to parse compileSDKVersion ()[0m

[34;1mEnsure dependencies[0m
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1minstall-missing-android-tools@1.0.2[0m                           | 3.48 sec |
+---+---------------------------------------------------------------+----------+
```

**Comment by alshari86** [comment]
Thanks for your looking into this, I also contacted bitrise online chat and the error is a warning and will not stop building.
In my build failure, there is another thing besides this is to remove "change work dir" in my work flow, but that step is in workflow by default when I created this app. So also need bitrise to look into

**Comment by godrei** [comment]
Hi @alshari86 ,
sorry for my late response and thanks for the issue report!

Could you please share your:

- banking/mobile/build.gradle
- /banking/misnapworkflow/build.gradle

build.gradle files contents?

It would help us to debug, why the analyzer failed.

**Comment by godrei** [comment]
Hi @alshari86 
we released a new version ([2.0.0](https://github.com/bitrise-steplib/steps-install-missing-android-tools/releases/tag/2.0.0)) , which uses gradle commands to install the missing components, this version should fix your issue.

I'll close this issue, but if you need more info/help just let us know and we'll reopen it!
